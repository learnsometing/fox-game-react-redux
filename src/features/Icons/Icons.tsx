import React from 'react';

const Icons: React.FC = () => (
  <div className="icons">
    <div className="icon highlighted fish-icon"></div>
    <div className="icon poop-icon"></div>
    <div className="icon weather-icon"></div>
  </div>
);

export default Icons;
