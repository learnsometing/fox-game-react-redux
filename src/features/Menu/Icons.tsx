import styled from 'styled-components';
import notSelected from './ICONS/NotSelected.png';
import highlighted from './ICONS/Hover.png';
import selected from './ICONS/Selected.png';

const Icons = styled.div`
  position: absolute;
  top: 572px;
  left: 194px;
  height: 67px;
  width: 374px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export default Icons;

interface IconProps {
  isHighlighted: boolean;
}

export const FishIcon = styled.div<IconProps>`
  background-image: ${(props) =>
    props.isHighlighted ? `url(${highlighted})` : `url(${notSelected})`};
  background-position: 0px;
  width: 56px;
  height: 65px;
  overflow: hidden;
  background-repeat: no-repeat;
  background-color: transparent;
  border-color: transparent;

  &:active {
    background-image: url(${selected});
  }
`;

export const PoopIcon = styled(FishIcon)`
  background-position: -162px 0px;
`;

export const WeatherIcon = styled(FishIcon)`
  background-position: -323px 0px;
`;
