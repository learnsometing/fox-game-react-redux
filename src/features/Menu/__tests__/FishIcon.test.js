import React from 'react';
import { FishIcon } from '../Icons';
import highlighted from '../ICONS/Hover.png';
import { create } from 'react-test-renderer';
import 'jest-styled-components';

describe('FishIcon', () => {
  it('should match the snapshot', () => {
    const fish = create(<FishIcon isHighlighted={false} />).toJSON();
    expect(fish).toMatchSnapshot();
  });

  it('should change the background image when isHighlighted', () => {
    const fish = create(<FishIcon isHighlighted={true} />).toJSON();
    expect(fish).toHaveStyleRule('background-image', `url(${highlighted})`);
  });
});
