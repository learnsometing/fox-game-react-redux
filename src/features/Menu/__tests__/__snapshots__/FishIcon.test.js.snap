// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FishIcon should match the snapshot 1`] = `
.c0 {
  background-image: url(NotSelected.png);
  background-position: 0px;
  width: 56px;
  height: 65px;
  overflow: hidden;
  background-repeat: no-repeat;
  background-color: transparent;
  border-color: transparent;
}

.c0:active {
  background-image: url(Selected.png);
}

<div
  className="c0"
/>
`;
